src/main/kotlin/no/nav/tilleggsstonader/sak/beregnfra/UtledBeregnFraDatoService.kt

```diff
@@ -0,0 +1,215 @@
+            erEndretFunksjon = ::erVedtaksperiodeEndret,
+        )
+
+    private fun erVilkårEndret(
+        vilkårNå: Vilkår,
+        vilkårTidligereBehandling: Vilkår,
+    ): Boolean =
+        vilkårNå.utgift != vilkårTidligereBehandling.utgift ||
+            // barnId er forskjellig for hver behandling, sjekker på ident
+            barnIdTilIdentMap[vilkårNå.barnId] != barnIdTilIdentMap[vilkårTidligereBehandling.barnId] ||
+            vilkårNå.erFremtidigUtgift != vilkårTidligereBehandling.erFremtidigUtgift ||
+            vilkårNå.type != vilkårTidligereBehandling.type ||
+            vilkårNå.resultat != vilkårTidligereBehandling.resultat ||
+            delvilkårErEndret(vilkårNå, vilkårTidligereBehandling)
+
+    private fun delvilkårErEndret(
+        vilkårNå: Vilkår,
+        vilkårTidligereBehandling: Vilkår,
+    ): Boolean = vilkårNå.delvilkårsett.utenVurderinger() != vilkårTidligereBehandling.delvilkårsett.utenVurderinger()
+
+    private fun erMålgruppeEllerAktivitetEndret(
+        vilkårperiode: GeneriskVilkårperiode<*>,
+        tidligereVilkårperiode: GeneriskVilkårperiode<*>,
+    ): Boolean =
+        vilkårperiode.resultat != tidligereVilkårperiode.resultat ||
+            vilkårperiode.type != tidligereVilkårperiode.type ||
+            vilkårperiode.faktaOgVurdering != tidligereVilkårperiode.faktaOgVurdering
+
+    private fun erVedtaksperiodeEndret(
+        vedtaksperiode: Vedtaksperiode,
+        tidligereVedtaksperiode: Vedtaksperiode,
+    ): Boolean =
+        vedtaksperiode.målgruppe != tidligereVedtaksperiode.målgruppe ||
+            vedtaksperiode.aktivitet != tidligereVedtaksperiode.aktivitet
+
+    private fun <T : Periode<LocalDate>> utledEndringIPeriode(
+        perioderNå: List<T>,
+        perioderTidligere: List<T>,
+        erEndretFunksjon: (T, T) -> Boolean,
+    ): LocalDate? {
+        val antallPerioder = min(perioderNå.size, perioderTidligere.size)
+        var i = 0
+        while (i < antallPerioder) {
+            val periodeNå = perioderNå[i]
+            val periodeTidligere = perioderTidligere[i]
+
+            if (periodeNå.fom != periodeTidligere.fom || erEndretFunksjon(periodeNå, periodeTidligere)) {
+                return minOf(periodeNå.fom, periodeTidligere.fom)
+            }
+            if (periodeNå.tom != periodeTidligere.tom) {
+                // Legger på en dag, da det først er fra dagen etter tom-datoen at det er en endring
+                return minOf(periodeNå.tom, periodeTidligere.tom).plusDays(1)
+            }
+
+            i++
+        }
+
+        return if (i < perioderNå.size) {
+            // Ny periode i ny behandling
+            perioderNå[i].fom
+        } else if (i < perioderTidligere.size) {
+            // Periode har blitt slettet i ny behandling
+            perioderTidligere[i].fom
+        } else {
+            // Ingen endringer i perioder
+            null
+        }
```

Comment:
Den `i`'en ble først litt forvirrende for meg. Ikke sikkert dette føles enklere.. men et alternativ iaf
```suggestion
        perioderNå.forEachIndexed { index, periodeNå ->
            if (index >= perioderTidligere.size) {
                return periodeNå.fom
            }
            val periodeTidligere = perioderTidligere[index]

            if (periodeNå.fom != periodeTidligere.fom || erEndretFunksjon(periodeNå, periodeTidligere)) {
                return minOf(periodeNå.fom, periodeTidligere.fom)
            }
            if (periodeNå.tom != periodeTidligere.tom) {
                // Legger på en dag, da det først er fra dagen etter tom-datoen at det er en endring
                return minOf(periodeNå.tom, periodeTidligere.tom).plusDays(1)
            }
        }

        return if (perioderNå.size < perioderTidligere.size) {
            // Periode har blitt slettet i ny behandling
            perioderTidligere[perioderNå.size].fom
        } else {
            // Ingen endringer i perioder
            null
        }
```